src/System.Management.Automation/utils/PsUtils.cs

```diff
@@ -59,13 +59,12 @@ internal static class PsUtils
         internal static ProcessModule GetMainModule(Process targetProcess)
         {
             int caughtCount = 0;
-            ProcessModule mainModule = null;
 
-            while (mainModule == null)
+            while (true)
             {
                 try
                 {
-                    mainModule = targetProcess.MainModule;
+                    return targetProcess.MainModule;
                 }
                 catch (System.ComponentModel.Win32Exception e)
```

Comment:
Should we catch `InvalidOperationException`? 
https://msdn.microsoft.com/en-us/library/system.diagnostics.process.mainmodule%28v=vs.110%29.aspx

Reply:
@daxian-dbw what do you think?

Reply:
I don't think we should catch `InvalidOperationException`. It seems to me we catch `Win32Exception` only to work around the flaw described in the summary.

Reply:
Why we should wait if process already exit? I'd re-throw immediately.

Reply:
`InvalidOperationException` will be thrown when process already exists, but we don't catch `InvalidOperationException` and thus no waiting in that case.

Reply:
Thanks for clarify!